### PR TITLE
SYS-3338 Drops OwnedNfts storage item

### PR DESCRIPTION
**Pull request by thadouk**
Adds a migration that drops OwnedNfts
Updates pallet version to use the new pallet versioning schema

### CODE REVIEW

File: pallets/nft-manager/src/lib.rs [hunk 1]
```diff
@@ -767,6 +758,68 @@ pub mod pallet {
...
+                    log::info!(
+                        "✅ Cleared '{}' entries from 'NftManager::OwnedNfts' storage prefix",
+                        res.unique
+                    );
+                    if res.maybe_cursor.is_some() {
+                        log::error!(
+                            "Storage prefix 'NftManager::OwnedNfts' is not completely cleared."
+                        );
+                    }
+                }
```
Comment (nahuseyoum) [inline pallets/nft-manager/src/lib.rs:792]: Can we replace this with `<OwnedNfts<T>>::kill()`?

File: pallets/nft-manager/src/lib.rs [hunk 2]
```diff
@@ -767,6 +758,68 @@ pub mod pallet {
...
+                        res.backend
+                    );
+                    log::info!(
+                        "✅ Cleared '{}' entries from 'NftManager::StorageVersion' storage prefix",
+                        res.unique
+                    );
+                    if res.maybe_cursor.is_some() {
+                        log::error!("Storage prefix 'StorageVersion' is not completely cleared.");
+                    }
+                }
```
Comment (nahuseyoum) [inline pallets/nft-manager/src/lib.rs:812]: I think this is duplicated code
```suggestion
                
```

Comment (thadouk) [reply pallets/nft-manager/src/lib.rs:812]: no its for the StorageVersion item, but need to update the variable name

File: pallets/nft-manager/src/lib.rs (hunk 1 above)
Comment (thadouk) [reply pallets/nft-manager/src/lib.rs:792]: The reason I opted for this was that it doesn't require keeping the storage item in the pallet code. But I can do it like that and remove the reference afterwards.

File: pallets/nft-manager/src/lib.rs [hunk 3]
```diff
@@ -121,7 +122,8 @@ pub mod pallet {
 
     #[pallet::pallet]
     #[pallet::generate_store(pub (super) trait Store)]
-    #[pallet::without_storage_info]
+    #[pallet::storage_version(STORAGE_VERSION)]
+    // #[pallet::without_storage_info]
```
[the remark this answers is not in the document]
Comment (thadouk) [reply pallets/nft-manager/src/lib.rs]: no, we want to generate the storage info. The main reason we didn't was that we weren't using bounded data structures.

**Review by nahuseyoum — APPROVED** [review]

**Review by ivan-cholakov — APPROVED** [review]

**Pull request merged**